Automattic/wp-calypso
# Command Palette: Wrap it with a context to prevent prop drilling

Fixes #87949

## Proposed Changes

* Wrap the `CommandPalette`  component with a context provider so the props needed on nested children can be accessed with the `useCommandPaletteContext` hook.
* Also add a `CommandMenuGroupContext` to handle the state props passed down to children.

## Testing Instructions

Open the PR preview and check that the command palette works as usual.

Note: The code review is easier by hiding the whitespace on the diff.

## Pre-merge Checklist
